pkg/oci/retry.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2025 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oci
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	retryLimit = 20
+	retryDelay = 500 * time.Millisecond
+)
+
+// errRetry hints that the function can safely be tried again after a small
+// time.
+var errRetry = errors.New("retry")
+
+// ErrRetryLimitExceeded indicates the function was retry too many times.
+var ErrRetryLimitExceeded = errors.New("retry limit exceeded")
```

Comment:
```suggestion
var errRetryLimitExceeded = errors.New("retry limit exceeded")
```

Reply:
I think it's fine to have this public. Callers could use it.